magma/syntax/sequential.py

```diff
@@ -364,50 +369,28 @@ def _sequential(
         comb_out_wiring=comb_out_wiring
     )
     tree = ast.parse(circuit_definition_str)
-    if "DefineRegister" not in defn_env:
-        tree = ast.Module([
-            ast.parse("from mantle import DefineRegister").body[0],
-        ] + tree.body)
-
-    circuit_def_constructor = ast_utils.compile_function_to_file(tree, 'make_' + cls.__name__, defn_env)
-    circuit_def = circuit_def_constructor(combinational_decorator)
-
-    if get_debug_mode() and getattr(circuit_def, "debug_info", False):
-        circuit_def.debug_info = debug_info(circuit_def.debug_info.filename,
-                                            circuit_def.debug_info.lineno,
-                                            inspect.getmodule(cls))
-
-    return circuit_def
-
-def sequential(
-        cls=None,
-        async_reset=None,
-        *,
-        decorators: typing.Optional[typing.Sequence[typing.Callable]] = None,
-        ):
-
-    exec(_SKIP_FRAME_DEBUG_STMT)
-    if async_reset is not None or decorators is not None:
-        assert async_reset is None or isinstance(async_reset, bool)
-        assert cls is None
-        if decorators is None:
-            decorators = ()
-
-        def wrapped(cls):
-            exec(_SKIP_FRAME_DEBUG_STMT)
-            nonlocal decorators
-            decorators = list(itertools.chain(decorators, [wrapped]))
-            wrapped_sequential = ast_utils.inspect_enclosing_env(
-                    _sequential,
-                    decorators=decorators)
-            combinational = m.circuit.combinational(decorators=decorators)
-            return wrapped_sequential(async_reset, cls, combinational)
-        return wrapped
-    else:
-        assert cls is not None
-        wrapped_sequential = ast_utils.inspect_enclosing_env(
-                _sequential,
-                decorators=[sequential]
-        )
-        combinational = m.circuit.combinational(decorators=[sequential])
-        return wrapped_sequential(True, cls, combinational)
+    if "DefineRegister" not in env:
```

Comment:
I know this is a bit paranoid but all of these should should be injecting with a free name then passing the generated names around.  e.g.:
```python
def gen_register_instances(def_reg_name, initial_value_map, async_reset):
    register_instances = []
    for name, (value, type_, eval_type, eval_value) in initial_value_map.items():
        if isinstance(eval_type, m.Kind):
            if isinstance(eval_type, m._BitKind):
                n = None
                if isinstance(eval_value, (bool, int)):
                    init = bool(eval_value)
                else:
                    assert eval_value.name.name in ["GND", "VCC"], eval_value.name
                    init = 0 if eval_value.name.name == "GND" else 1
            else:
                n = len(eval_type)
                init = int(eval_value)
            register_instances.append(f"{name} = {def_reg_name}({n}, init={init}, has_async_reset={async_reset})()")  # noqa
        else:
            value = astor.to_source(value).rstrip()
            register_instances.append(f"{name} = {value}")
    return register_instances
```